#! /bin/sh /usr/share/dpatch/dpatch-run
## build_fix.patch by Stefan Lippers-Hollmann <s.l-h@gmx.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: port to debian.

@DPATCH@
diff -Nrup --exclude=debian --exclude='*stamp' kudzu-kanotix-1.2.9.orig/pci.c kudzu-kanotix-1.2.9/pci.c
--- kudzu-kanotix-1.2.9.orig/pci.c	2005-10-02 16:41:58.000000000 +0200
+++ kudzu-kanotix-1.2.9/pci.c	2005-10-02 16:44:12.303732500 +0200
@@ -258,7 +258,7 @@
 #endif	
 	unsigned int devtype, command;
 	
-	devtype = p->device_class;
+	devtype = config[PCI_CLASS_DEVICE+1] << 8 | config[PCI_CLASS_DEVICE];
 	if (p->irq || pciToKudzu(devtype) != CLASS_VIDEO) {
 		return 0;
 	}
@@ -388,7 +388,7 @@ struct device * pciProbe(enum deviceClas
 		    subvend = config[PCI_SUBSYSTEM_VENDOR_ID+1] << 8 | config[PCI_SUBSYSTEM_VENDOR_ID];
 		    subdev = config[PCI_SUBSYSTEM_ID+1] << 8 | config[PCI_SUBSYSTEM_ID];
 		}
-		pci_fill_info(p, PCI_FILL_IDENT | PCI_FILL_CLASS | PCI_FILL_IRQ | PCI_FILL_BASES | PCI_FILL_ROM_BASE | PCI_FILL_SIZES);
+		pci_fill_info(p, PCI_FILL_IDENT | PCI_FILL_IRQ | PCI_FILL_BASES | PCI_FILL_ROM_BASE | PCI_FILL_SIZES);
 		bustype = PCI_NORMAL;
 	        for (bridgenum=0; cardbus_bridges[bridgenum]; bridgenum++) {
 			if (p->bus == cardbus_bridges[bridgenum])
@@ -401,13 +401,13 @@ struct device * pciProbe(enum deviceClas
 		dev->subDeviceId = subdev;
 		dev->pciType = bustype;
 		asprintf(&t,"v%08Xd%08Xsv%08Xsd%08Xbc%02Xsc%02Xi%02x",p->vendor_id,p->device_id,
-			 subvend, subdev,(u_int8_t)(p->device_class >> 8),(u_int8_t)(p->device_class),config[PCI_CLASS_PROG]);
+			 subvend, subdev,(u_int8_t)((config[PCI_CLASS_DEVICE+1] << 8 | config[PCI_CLASS_DEVICE]) >> 8),(u_int8_t)(config[PCI_CLASS_DEVICE+1] << 8 | config[PCI_CLASS_DEVICE]),config[PCI_CLASS_PROG]);
 		drv = aliasSearch(aliases, "pci", t);
 		x_drv = aliasSearch(aliases, "pcivideo", t);
 		free(t);
 		if (drv)
 			dev->driver = strdup(drv);
-		devtype = p->device_class;
+		devtype = config[PCI_CLASS_DEVICE+1] << 8 | config[PCI_CLASS_DEVICE];
 		if (x_drv) {
 			dev->classprivate = strdup(x_drv);
 		}
